packages/file/lib/src/backends/memory/memory_file.dart

```diff
@@ -314,12 +314,28 @@ class _FileSink implements io.IOSink {
     io.FileMode mode,
     Encoding encoding,
   ) {
-    Future<FileNode> node = Future<FileNode>.microtask(() {
-      FileNode node = file._resolvedBackingOrCreate;
+    FileNode node;
+    Exception deferredException;
+
+    // Resolve the backing immediately to ensure that the [FileNode] we write
```

Comment:
We should probably check that the style is posix style here (see https://github.com/google/file.dart/blob/master/packages/file/lib/src/backends/memory/memory_file_system.dart#L46).  I think it'd probably be fairly simple given that to implement the windows type behavior - wdyt?

Reply:
I agree that checking `style` would be a good way to configure the behavior.  I don't think here would be the right place to do it.  Windows typically wouldn't allow renaming/moving/deleting files with open handles in the first place, so I don't know what (if anything) we should do differently here. (I also currently don't have a Windows machine handy with Dart set up to verify `dart:io`'s `File`'s behavior there.)

Reply:
Do we have a bug tracking work on that?

Reply:
Not yet.  I will file one.